libraries/AP_Math/control.cpp

```diff
@@ -387,7 +387,7 @@ bool limit_accel_xy(const Vector2f& vel, Vector2f& accel, float accel_max)
             if (accel_cross.limit_length(accel_max)) {
                 accel_dir = 0.0;
             } else {
-                float accel_max_dir = safe_sqrt(sq(accel_max) - accel_cross.length_squared());
+                float accel_max_dir = safe_sqrt(MAX(0, sq(accel_max) - accel_cross.length_squared()));
```

Comment:
```suggestion
                // limit_length can't absolutely guarantee this subtraction won't be slightly negative
                float accel_max_dir = safe_sqrt(MAX(0, sq(accel_max) - accel_cross.length_squared()));
```

Reply:
Actually, it would probably be more efficient to just calculate the two terms of the subtraction and compare them.

Reply:
Done